components/gateway/internal/auditlog/client.go

```diff
@@ -76,22 +76,37 @@ func (c *Client) LogSecurityEvent(ctx context.Context, event model.SecurityEvent
 
 func (c *Client) sendAuditLog(ctx context.Context, req *http.Request) error {
 	logger := log.C(ctx)
+	responseStatusCode, err := c.sendAuditLogInternal(ctx, req)
+	if responseStatusCode != http.StatusCreated || err != nil {
+		logger.Infof("Will retry auditlog request.")
+		responseStatusCode, err = c.sendAuditLogInternal(ctx, req)
```

Comment:
Due to the fact that we make a `POST` request, we can't reuse the same `req` object for the second attempt (the retry) because the request body will be read from the first call and here we will try to make a `POST` request with an empty body which will result in a similar error:

` Post <url>: http: <someInfo> with Body length 0`

I faced the same issue and the most straightforward solution is either to build the request object before each http call or in order to reuse the `req` object to set the request body before each call